connect/telegram_bot/bot_runner.py

```diff
@@ -0,0 +1,80 @@
+import logging
+import sys
+
+from django.core.exceptions import ImproperlyConfigured
+from django.utils.module_loading import import_string
+
+from telegram import ReplyKeyboardRemove
+from telegram.ext import Updater
+
+from django.conf import settings
+
+
+class BotRunner(object):  # pragma: no cover
+
+    def __init__(self):
+        self._get_settings()
+
+    def get_conversation_handler(self, conv_class):
+        try:
+            class_def = import_string(conv_class)
+
+            return class_def(
+                logger=logging.getLogger('django_telegram_bot'),
+                model_datetime_property=self.model_property,
+                suffix=self.suffix,
+            ).get_conversation_handler()
+        except ImportError:
+            return None
+
+    @staticmethod
+    def error_callback(update, context):
+        update.message.reply_text(
+            f'*ERR*: ``` {context.error} ```',
+            parse_mode='markdown',
+            reply_markup=ReplyKeyboardRemove(selective=True),
+            reply_to_message_id=update.message.message_id,
+            api_kwargs={'chat_id': update.message.chat.id},
+        )
+
+    def _get_settings(self):
+        try:
+            self.token = settings.TELEGRAM_BOT['TOKEN']
+            self.suffix = settings.TELEGRAM_BOT['COMMANDS_SUFFIX']
+            self.model_property = settings.TELEGRAM_BOT['HISTORY_LOOKUP_MODEL_PROPERTY']
+            self.conv_classes = settings.TELEGRAM_BOT['CONVERSATIONS']
+        except (KeyError, AttributeError, ImproperlyConfigured) as e:
+            exit(f'Error getting settings: {str(e)}')
```

Comment:
ImproperlyConfigured is not raised here. I think is better to catch KeyError and AttributeError and raise ImproperlyConfigured with a meaningful message.
Probably is better to do so in the AppConfig class for your app (that is missed).

Reply:
fixed